nighthawk/README.md

```diff
@@ -0,0 +1,154 @@
+# Nighthawk
+
+*A benchmarking tool based on Envoy*
+
+## Current state
+
+The nighthawk client supports HTTP/1.1 and HTTP/2 over http and https. (NOTE: https certificates are not yet validated).
+
+## Prerequisites
+
+### Ubuntu
+
+1. Install the latest version of [Bazel](https://bazel.build/versions/master/docs/install.html) in your environment.
+
+2. Install external dependencies libtool, cmake, ninja, realpath and curl libraries separately.
+On Ubuntu, run the following command:
+
+``` bash
+
+# For Nighthawk / Envoy. 
+# Realpath may give an error, when it does that you probably already have it.
+
+sudo apt-get install \
+   libtool \
+   cmake \
+   realpath \
+   clang-format-7 \
+   automake \
+   ninja-build \
+   curl \
+   unzip
+
+# For tools/stats.py
+sudo apt-get install python3 python3-pip
+sudo pip3 install hdrhistogram jsonpickle
+```
+
+## Building and testing Nighthawk
+```zsh
+# TODO(oschaaf): Collect and list prerequisites. 
+# Currently mostly the same as Envoy.
+
+# build it
+bazel build //:nighthawk_client
```

Comment:
Should this be `-c opt`? That would likely make a huge difference in performance.